### PR TITLE
support compare traces in comparetest

### LINKED ISSUE #17414: [internal/scrapertest] Add traces support
The [scrapertest](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/internal/scrapertest) package contains functions for comparing metrics and logs only

We should also extend this package to include similar functions for traces (ReadTraces, WriteTrace, and CompareTraces).

### PR DESCRIPTION
**Pull request by fatsheep9146**
Signed-off-by: Ziqi Zhao <zhaoziqi9146@gmail.com>

**Description:** 
Fix #17414

### CODE REVIEW

File: internal/comparetest/golden/golden.go [hunk 1]
```diff
@@ -96,3 +97,33 @@ func WriteLogs(filePath string, logs plog.Logs) error {
...
+	if err != nil {
+		return ptrace.Traces{}, err
+	}
+
+	unmarshaler := ptrace.JSONUnmarshaler{}
+	return unmarshaler.UnmarshalTraces(b)
+}
+
+// WriteTraces writes a ptrace.Traces to the specified file
+func WriteTraces(filePath string, traces ptrace.Traces) error {
```
Comment (djaglowski) [inline internal/comparetest/golden/golden.go:113]: Please see updated implementation of `WriteMetrics`. We should force users to pass in `*testing.T` so we can fail tests that use this function.

**Comment by djaglowski** [comment]
Looks like you may have had a problem rebasing. Lots of unrelated changes.

**Comment by fatsheep9146** [comment]
> Looks like you may have had a problem rebasing. Lots of unrelated changes.

yes, thanks for reminding me. I will check about this. @djaglowski

**Comment by fatsheep9146** [comment]
@djaglowski @dmitryax The pr is ready to be reviewed.
@open-telemetry/collector-contrib-approvers

File: internal/comparetest/options.go [hunk 2]
```diff
@@ -33,6 +34,11 @@ type LogsCompareOption interface {
 	applyOnLogs(expected, actual plog.Logs)
 }
 
+// TracesCompareOption can be used to mutate expected and/or actual traces before comparing.
+type TracesCompareOption interface {
```
Comment (dmitryax) [inline internal/comparetest/options.go:38]: Can you please add traces support in `IgnoreResourceAttributeValue` option as a follow-up PR?

File: internal/comparetest/traces_test.go [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+func TestCompareTraces(t *testing.T) {
+	tcs := []struct {
+		name           string
+		compareOptions []TracesCompareOption
+		withoutOptions expectation
+		withOptions    expectation
+	}{
+		{
+			name: "equal",
+		},
```
Comment (dmitryax) [inline internal/comparetest/traces_test.go:35]: Once you have a chance, can you please add more test cases for traces?

**Review by dmitryax — APPROVED** [review]
LGTM. Thanks, @fatsheep9146!

File: internal/comparetest/options.go (hunk 2 above)
Comment (fatsheep9146) [reply internal/comparetest/options.go:38]: I will create a pr to record the expected enhancements of traces. @dmitryax

**Pull request merged**